scripts/extensions/availability-manager/src/correspondent-availability/day-view.tsx

```diff
@@ -0,0 +1,178 @@
+    participantIds: Set<IUser['_id']> | null;
+}
+
+export class DayView extends React.PureComponent<IProps, IState> {
+    constructor(props: IProps) {
+        super(props);
+
+        this.state = {
+            participantIds: null,
+        };
+    }
+
+    componentDidMount(): void {
+        fetchParticipants().then((items) => {
+            this.setState({participantIds: items});
+        });
+    }
+
+    render() {
+        const {filters} = this.props;
+        const {participantIds} = this.state;
+
+        if (participantIds == null) { // loading
+            return null;
+        }
+
+        const users: {[key: string]: IUser} = superdesk.entities.users.getAllUsers();
+
+        return (
+            <WithAvailabilityRecords dateFrom={filters.date} dateTo={filters.date} filters={filters}>
+                {({byUserByDateFiltered}) => {
+                    const records: Array<IAvailabilityRecord> = [];
+
+                    for (const byDate of Object.values(byUserByDateFiltered)) {
+                        for (const _records of Object.values(byDate)) {
+                            records.push(..._records);
+                        }
+                    }
+
+                    const recordsByUser: {[userId: string]: IAvailabilityRecord} = keyBy(records, ({user}) => user);
+                    const recordsSorted = sortAvailabilityRecords(records);
+                    const usersWithRecord = recordsSorted.map(({user}) => user);
+                    const usersWithoutRecord = difference(
+                        Array.from(participantIds),
+                        usersWithRecord,
+                    ).sort((a, b) => compareUsersByName(users[a], users[b]));
+
+                    const participantsIds = filterParticipants({
+                        participantIds: [
+                            ...usersWithRecord,
+                            ...usersWithoutRecord,
+                        ],
+                        days: [filters.date],
+                        filters: filters,
+                        byUserByDateFiltered,
+                    });
+
+                    return (
+                        <BoxedList style={{padding: 'var(--space--2)'}}>
+                            {participantsIds.map((participantId) => {
+                                const record = recordsByUser[participantId] as IAvailabilityRecord | null;
+                                const user = users[participantId];
+
+                                return (
+                                    <BoxedListItem
+                                        key={participantId}
+                                        type={(() => {
+                                            if (record == null) {
+                                                return 'default';
+                                            }
+
+                                            switch (record.status) {
```

Comment:
looks like that logic is duplicated in https://github.com/superdesk/superdesk-client-core/pull/4813/files#diff-a4368b585f83eeb3681d18f74857ee73a05eefb16b713c8ca8a6247e98b033c2R127-R137